tensorflow_lite_support/cc/test/task/processor/image_preprocessor_test.cc

```diff
@@ -0,0 +1,147 @@
+#include "tensorflow_lite_support/cc/port/status_matchers.h"
+#include "tensorflow_lite_support/cc/task/core/task_api_factory.h"
+#include "tensorflow_lite_support/cc/task/core/task_utils.h"
+#include "tensorflow_lite_support/cc/task/core/tflite_engine.h"
+#include "tensorflow_lite_support/cc/task/vision/utils/frame_buffer_common_utils.h"
+#include "tensorflow_lite_support/cc/test/test_utils.h"
+#include "tensorflow_lite_support/examples/task/vision/desktop/utils/image_utils.h"
+
+namespace tflite {
+namespace task {
+namespace processor {
+namespace {
+
+using ::testing::ElementsAreArray;
+using ::testing::HasSubstr;
+using ::testing::Optional;
+using ::tflite::support::kTfLiteSupportPayload;
+using ::tflite::support::StatusOr;
+using ::tflite::support::TfLiteSupportStatus;
+using ::tflite::task::JoinPath;
+using ::tflite::task::core::PopulateTensor;
+using ::tflite::task::core::TaskAPIFactory;
+using ::tflite::task::core::TfLiteEngine;
+using ::tflite::task::vision::DecodeImageFromFile;
+using ::tflite::task::vision::FrameBuffer;
+using ::tflite::task::vision::ImageData;
+
+constexpr char kTestDataDirectory[] =
+    "tensorflow_lite_support/cc/test/testdata/task/vision/";
+
+constexpr char kDilatedConvolutionModelWithMetaData[] = "dilated_conv.tflite";
+
+StatusOr<ImageData> LoadImage(std::string image_name) {
+  return DecodeImageFromFile(
+      JoinPath("./" /*test src dir*/, kTestDataDirectory, image_name));
+}
+
+class DynamicInputTest : public tflite_shims::testing::Test {
+ public:
+  void SetUp() {
+    engine_ = absl::make_unique<TfLiteEngine>();
+    engine_->BuildModelFromFile(JoinPath("./", kTestDataDirectory,
+                                         kDilatedConvolutionModelWithMetaData));
+    engine_->InitInterpreter();
+
+    SUPPORT_ASSERT_OK_AND_ASSIGN(preprocessor_,
+                                 ImagePreprocessor::Create(engine_.get(), {0}));
+  }
+
+ protected:
+  std::unique_ptr<ImagePreprocessor> preprocessor_ = nullptr;
+  std::unique_ptr<TfLiteEngine> engine_ = nullptr;
+};
+
+// See if input tensor dims signature for height and width is -1
+// because it is so in the model.
+TEST_F(DynamicInputTest, InputHeightAndWidthMutable) {
+  const TfLiteIntArray *input_dims_signature =
+      engine_->GetInputs()[0]->dims_signature;
+  EXPECT_EQ(input_dims_signature->data[1], -1);
+  EXPECT_EQ(input_dims_signature->data[2], -1);
+}
+
+// See if output tensor has been re-dimmed as per the input
+// tensor. Expected shape: (1, input_height, input_width, 16).
+TEST_F(DynamicInputTest, OutputDimensionCheck) {
+  SUPPORT_ASSERT_OK_AND_ASSIGN(ImageData image, LoadImage("burger.jpg"));
+  std::unique_ptr<FrameBuffer> image_frame_buffer = CreateFromRgbRawBuffer(
+      image.pixel_data, FrameBuffer::Dimension{image.width, image.height});
+
+  preprocessor_->Preprocess(*image_frame_buffer);
+  absl::Status status = engine_->interpreter_wrapper()->InvokeWithoutFallback();
+  EXPECT_TRUE(status.ok());
```

Comment:
This can be merged into one line, such as:
SUPPORT_ASSERT_OK(engine_->interpreter_wrapper()->InvokeWithoutFallback());